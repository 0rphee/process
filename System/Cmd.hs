-----------------------------------------------------------------------------
-- |
-- Module      :  System.Cmd
-- Copyright   :  (c) The University of Glasgow 2001
-- License     :  BSD-style (see the file libraries/base/LICENSE)
--
-- Maintainer  :  libraries@haskell.org
-- Stability   :  provisional
-- Portability :  portable
--
-- Executing an external command.
--
-- This module provides a simple interface for executing external commands.
-- For a more complex, but more powerful, interface, see the "System.Process"
-- module.
--
-----------------------------------------------------------------------------

-- later: {-# DEPRECATED "Use System.Process instead" #-}
module System.Cmd
    ( system,        -- :: String -> IO ExitCode
      rawSystem,     -- :: FilePath -> [String] -> IO ExitCode
    ) where

import System.Process
