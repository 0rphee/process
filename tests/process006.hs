module Main where

import Control.Concurrent
import System.IO
import System.Process
import Control.Monad
import Control.Exception

main :: IO ()
main = do 
  print =<< readProcess "cat" [] "yan\ntan\tether\n"
  print =<< readProcessWithExitCode "cat" [] "yan\ntan\tether\n"
  print =<< readProcessWithExitCode "ls" ["/does/not/exist"] ""
  print =<< (try $ readProcess "ls" ["/does/not/exist"] "")
