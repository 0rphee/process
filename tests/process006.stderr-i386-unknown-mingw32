ls: cannot access /does/not/exist: No such file or directory
