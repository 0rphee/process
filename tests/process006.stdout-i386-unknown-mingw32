ls: cannot access /does/not/exist: No such file or directory
"yan\ntan\tether\n"
(ExitSuccess,"yan\ntan\tether\n")
(ExitFailure 2,"")
Left readProcess: ls "/does/not/exist" (exit 2): failed
