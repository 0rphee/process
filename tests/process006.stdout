"yan\ntan\tether\n"
(ExitSuccess,"yan\ntan\tether\n","")
(ExitFailure 2,"","ls: /does/not/exist: No such file or directory\n")
Left readProcess: ls "/does/not/exist" (exit 2): failed
